src/esoh/ingest/main.py

```diff
@@ -0,0 +1,47 @@
+from esoh.ingest.send_mqtt import mqtt_connection
+from esoh.ingest.messages import load_files, build_message
+
+
+import xarray as xr
+
+
+class ingest_to_pipeline():
+    """
+    This class should be the main interaction with this python package.
+    Should accept paths or objects to pass on to the datastore and mqtt broker.
+    """
+
+    def __init__(self, mqtt_conf: dict, uuid_prefix: str, testing: bool = False):
+        self.uuid_prefix = uuid_prefix
+
+        if testing:
+            return
+
+        self.mqtt = mqtt_connection(mqtt_conf["host"], mqtt_conf["topic"])
+
+    def ingest_message(self, message: [str, object], input_type: str = None):
+        if not input_type:
+            input_type = self.decide_input_type(message)
+
+        return self.build_message(message, input_type)
+
+    def decide_input_type(self):
+        pass
+
+    def build_messages(self, message: [str, object], input_type: str = None):
+        match input_type:
```

Comment:
Why match case, isn't if sufficient and or better (one less indent)?

Reply:
The reason I chose to use this here is for ease of adding more filetypes at a later stage, and then making it easier to read when this list gets longer. I do agree that when there is only one filetype implemented there is no need to use match-case